api/v1beta1/manila_types.go

```diff
@@ -143,6 +147,31 @@ func init() {
 	SchemeBuilder.Register(&Manila{}, &ManilaList{})
 }
 
+// DBPurge struct is used to model the parameters exposed to the Manila API CronJob
+type DBPurge struct {
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default=30
```

Comment:
The kuttl failure [1] reveals a problem with initializing the DBPurge structure. 

[1] https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openstack-k8s-operators_manila-operator/170/pull-ci-openstack-k8s-operators-manila-operator-main-manila-operator-build-deploy-kuttl/1719630065320333312#1:build-log.txt%3A4201

Even though we want kubebuilder to provide the default values, they aren't set unless the CR explicitly includes dbPurge in the spec:
```
spec:
  dbPurge: {}
```
I did some local tests on the PR code, and changing the kubebuilder validation to Required and/or removing "omitempty" had no effect. I believe this is a side effect of having a defaulting webhook, which makes k8s think the CR has been full initialized, and so it doesn't apply the kubebuilder values.

I don't think it will be sufficient to update *every* manila CR sample to explicitly add the "dbPurge: {}" entry. I think the best way forward is to enhance the webhook code to do something like this:
```
func (spec *ManilaSpec) Default() {
	if (spec.DBPurgeSpec.Age == 0) && (spec.DBPurgeSpec.Schedule == "") {
		spec.DBPurgeSpec.Age = DBPurgeDefaultAge
		spec.DBPurgeSpec.Schedule = DBPurgeDefaultSchedule
	}
```

Reply:
ack and thanks for the feedback. 
I see pretty much the same thing looking at the CI execution and I think it's a good practice using webhooks. By doing this we can check (and default) the user input not only when the CR is applied for the first time, but also during the entire lifecycle of the component.
Updating the patch and adding this extra snippet.